pkg/networkpolicy/multinetegressrule.go

```diff
@@ -88,6 +122,65 @@ func (builder *EgressRuleBuilder) WithPeerPodSelector(podSelector metav1.LabelSe
 	return builder
 }
 
+// WithPeerNamespaceSelector appends new item with only NamespaceSelector into To Peer list.
+func (builder *EgressRuleBuilder) WithPeerNamespaceSelector(nsSelector metav1.LabelSelector) *EgressRuleBuilder {
+	glog.V(100).Infof("Adding peer namespace selector %v to EgressRule", nsSelector)
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.definition.To = append(builder.definition.To, v1beta1.MultiNetworkPolicyPeer{NamespaceSelector: &nsSelector})
+
+	return builder
+}
+
+// WithCIDR edits last item's IPBlock on Egress/To list or adds new item with only IPBlock into
+// Egress/To list if the Egress/To list is empty.
+func (builder *EgressRuleBuilder) WithCIDR(cidr string, except ...[]string) *EgressRuleBuilder {
+	glog.V(100).Infof("Adding peer CIDR %s to Egress Rule", cidr)
+
+	if len(except) != 0 {
+		glog.V(100).Infof("Adding CIDR except %v to Egress Rule", except[0])
+	}
+
+	_, _, err := net.ParseCIDR(cidr)
+
+	if err != nil {
+		glog.V(100).Infof("Invalid CIDR %s", cidr)
+
+		builder.errorMsg = fmt.Sprintf("invalid CIDR argument %s", cidr)
+
+		return builder
+	}
+
+	builder.definition.To = append(builder.definition.To, v1beta1.MultiNetworkPolicyPeer{})
+
+	// Append IPBlock config to the previously added Peer
+	builder.definition.To[len(builder.definition.To)-1].IPBlock = &v1beta1.IPBlock{CIDR: cidr}
+
+	if len(except) > 0 {
+		builder.definition.To[len(builder.definition.To)-1].IPBlock.Except = except[0]
+	}
```

Comment:
I'm fine approving as is, but it may be simpler to build the v1beta1.MultiNetworkPolicyPeer struct before appending